### PR TITLE
package ollama-bin added

### PR DESCRIPTION
**Pull request by maixnor**
## Add ollama-bin package

a tool to manage open-source LLMs on your machine. 

similar to the existing `ollama` package at `pkgs/tools/misc/ollama`.

https://ollama.ai

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/ollama-bin/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m755 -D $src $out/bin/ollama
+  '';
+
+  meta = with lib; {
+    homepage = "https://ollama.ai";
+    description = "Get up and running with large language models, locally";
+    platforms = platforms.linux;
```
Comment (jcdickinson) [inline pkgs/tools/misc/ollama-bin/default.nix:33]: the `maintainers` meta is required, even if empty.

**Review by jcdickinson — CHANGES_REQUESTED** [review]
Update the subject line of the commit, and the title of the PR to:

```
ollama-bin: init at 0.1.8
```

**Comment by markuskowa** [comment]
Since `ollama` is already packaged in its source version, what is the motivation to package it as a binary version? From-source builds are in general preferred.

**Comment by maixnor** [comment]
The motivation is to have as well the version directly from the maintainers. This might be useful when breaking changes in the build step take longer to implement or if one wants to compare the functionality to the "official" release.

Den 11 nov. 2023 13:35 skrev markuskowa ***@***.***>:

Since ollama is already packaged in its source version, what is the motivation to package it as a binary version? From-source builds are in general preferred.

**Comment by AndersonTorres** [comment]
> The motivation is to have as well the version directly from the maintainers.

Irrelevant.
The upstream maintainers already provide the source code, and Nixpkgs have a mature framework for building Go packages.
Also, Nix has its own cache servers. No need to duplicate stuff.

> This might be useful when breaking changes in the build step take longer to implement

In other words, if Nixpkgs maintainers are not quick enough to update the expression?

> or if one wants to compare the functionality to the "official" release.

It can be done externally. No need to maintain otherwise identical stuff.

I suggest you to create a NUR ovelay.

**Comment by maixnor** [comment]
Thank you for the hint of NUR overlay. I will look into that. Closing this PR

**Pull request closed without merging**